src/main/java/com/yahoo/bullet/record/SimpleBulletRecord.java

```diff
@@ -0,0 +1,146 @@
+/*
+ *  Copyright 2018, Oath Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.record;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A simple implementation of {@link BulletRecord}.
+ */
+public class SimpleBulletRecord extends BulletRecord {
+    private static final long serialVersionUID = -4045166645513428587L;
+
+    // Exposed for testing.
+    @Setter(AccessLevel.PACKAGE)
+    private Map<String, Object> data = new HashMap<>();
+
+    @Override
+    protected BulletRecord set(String field, Object object) {
+        Objects.requireNonNull(field);
+        data.put(field, object);
+        return this;
+    }
+
+    @Override
+    public Object get(String field) {
+        return data.get(field);
+    }
+
+    @Override
+    public boolean hasField(String field) {
+        return data.containsKey(field);
+    }
+
+    @Override
+    public int fieldCount() {
+        return data.size();
+    }
+
+    @Override
+    public Object getAndRemove(String field) {
+        return hasField(field) ? data.remove(field) : null;
+    }
+
+    @Override
+    public BulletRecord remove(String field) {
+        if (hasField(field)) {
+            data.remove(field);
+        }
+        return this;
+    }
+
+    @Override
+    public Iterator<Pair<String, Object>> iterator() {
+        return new Iterator<Pair<String, Object>>() {
+            Iterator<Map.Entry<String, Object>> entries = data.entrySet().iterator();
+
+            @Override
+            public boolean hasNext() {
+                return entries.hasNext();
+            }
+
+            @Override
+            public Pair<String, Object> next() {
+                Map.Entry<String, Object> entry = entries.next();
+                return new ImmutablePair<>(entry.getKey(), entry.getValue());
+            }
+        };
+    }
+
+    @Override
+    public String toString() {
```

Comment:
The next few methods are from AvroBulletRecord. Is there a way to not duplicate them?